src/libvx.c

```diff
@@ -1717,22 +1218,15 @@ vx_error vx_frame_transfer_data(const vx_video* video, vx_frame* frame)
 		}
 	}
 	else {
-		av_log(NULL, AV_LOG_WARNING, "A frame did not contain either audio or video data and the buffer could not be transferred.\n");
+		av_log(NULL, AV_LOG_WARNING, "A frame did not contain either audio or video data so the buffer could not be transferred\n");
 	}
 
 	// Frame properties that may have been updated after filtering
-	if (vx_frame_properties_from_metadata(frame, av_frame) != VX_ERR_SUCCESS) {
+	if ((ret = vx_frame_properties_from_metadata(frame, av_frame)) != VX_ERR_SUCCESS) {
 		goto cleanup;
 	}
 
-	return VX_ERR_SUCCESS;
-
 cleanup:
-	if (av_frame) {
-		av_frame_unref(av_frame);
-		av_frame_free(&av_frame);
```

Comment:
If an error occurs and this point is reached then the frame should not be cleared here, it is cleaned up when the video is closed.